subsys/prefetch/src/main/java/org/commonjava/indy/subsys/prefetch/PrefetchManager.java

```diff
@@ -54,42 +53,83 @@
     @Inject
     private PrefetchConfig config;
 
-    @WeftManaged
-    @ExecutorConfig( named = "prefetch-worker", threads = 5, priority = 1 )
     @Inject
-    private Executor executor;
+    private SpecialPathManager specialPathManager;
 
-    @Inject
-    private Instance<ContentListBuilder> listBuilders;
+    private volatile boolean stopped;
+
+    private final ExecutorService prefetchExecutor =
```

Comment:
We need to make these `@Inject @WeftManaged @ExecutorConfig(..)`, so we can manage their sizes in conf.d/threadpools.conf. This will also allow us to report metrics about their use, once we have Weft updated to handle that.

Reply:
I used it in old way, but I found a very weird thing here: if I used a weft-managed thread pool, the running thread is not from this executor pool, but through the calling thread(ArtifactStorePostUpdateEvent thread), which is shown in the log. Not sure if this is caused by weft Context switch in ThreadContext or anywhere else. 
I'll change it back to weft way, but maybe it is still need for further considerations. Will give a comment of TODO here for marking.